web_src/js/components/DiffCommitSelector.vue

```diff
@@ -0,0 +1,149 @@
+        <div class="gt-ellipsis">
+          {{ locale.show_all_commits }}
+        </div>
+        <div class="gt-ellipsis text light-2">
+          {{ locale.stats_num_commits }}
+        </div>
+      </div>
+
+      <!-- only show the show changes since last review if there is a review AND we are commits ahead of the last review -->
+      <div v-if="lastReviewCommitSha != null && commitsSinceLastReview > 0" class="vertical item gt-df gt-fc gt-gap-1 gt-border-secondary-top" @click="changesSinceLastReviewClick()">
+        <div class="gt-ellipsis">
+          {{ locale.show_changes_since_your_last_review }}
+        </div>
+        <div class="gt-ellipsis text light-2">
+          {{ commitsSinceLastReview }} commits
+        </div>
+      </div>
+
+      <span class="info gt-border-secondary-top">{{ locale.select_commit_hold_shift_for_range }}</span>
+
+      <template v-for="commit in commits" :key="commit.id">
+        <div class="vertical item gt-df gt-fr gt-gap-2 gt-border-secondary-top" :class="{selected: commit.selected}" @click.exact="commitClicked(commit.id)" @click.shift.exact.stop.prevent="commitClickedShift(commit)">
+          <div class="gt-f1 gt-df gt-fc gt-gap-1">
+            <div class="gt-ellipsis commit-list-summary">
+              {{ commit.summary }}
+            </div>
+            <div class="gt-ellipsis text light-2">
+              {{ commit.committerOrAuthorName }}
+              <span class="text right">
+                <relative-time class="time-since" prefix="" :datetime="commit.time" data-tooltip-content data-tooltip-interactive="true">{{ commit.time }}</relative-time>
+              </span>
+            </div>
+          </div>
+          <div class="gt-mono">
+            {{ commit.id }}
+          </div>
+        </div>
+      </template>
+    </div>
+  </div>
+</template>
+
+<script>
+import {SvgIcon} from '../svg.js';
+
+export default {
+  components: {SvgIcon},
+  data: () => {
+    return {
+      menuVisible: false,
+      locale: {},
+      commits: [],
+      hoverActivated: false,
+      lastReviewCommitSha: null
+    };
+  },
+  computed: {
+    commitsSinceLastReview() {
+      if (this.lastReviewCommitSha) {
+        return this.commits.length - this.commits.findIndex((x) => x.id === this.lastReviewCommitSha) - 1;
+      }
+      return 0;
+    },
+    queryParams() {
+      return this.$el.parentNode.getAttribute('data-queryparams');
+    },
+    issueLink() {
+      return this.$el.parentNode.getAttribute('data-issuelink');
+    }
+  },
+  methods: {
+    /** Opens our menu, loads commits before opening */
+    async toggleMenu() {
+      // load our commits when the menu is not yet visible (it'll be toggled after loading)
+      // and we got no commits
+      if (this.commits.length === 0 && this.menuVisible === false) {
+        await this.fetchCommits();
+      }
+      this.menuVisible = !this.menuVisible;
```

Comment:
If the network/response is not fast (eg: 2second), then when a user clicks the dropdown, they sees nothing in 2 seconds.

Ideally, when the user clicks the dropdown, the menu should be shown immediately.

Reply:
i'll add a loading animation inside the dropdown.